posts/2021-11-18-til-cleaner-queries-with-ecto-map.md

```diff
@@ -0,0 +1,75 @@
+%{
+author: "Yuri Oliveira",
+author_link: "https://github.com/yuriploc",
+tags: ["ecto"],
+date: ~D[2021-11-18],
+title: "TIL: Cleaner queries with Ecto `map`",
+excerpt: """
+How learning more about the Ecto.Query module can help you write cleaner select queries.
+"""
+}
+
+---
+
+# TIL: Cleaner queries with Ecto `map`
+
+It's not unusual for database queries to avoid select * when we can use an index (say, :id) and have a performance gain.
```

Comment:
I don't think its necessarily more performant from a DB query execution perspective to be explicit vs. using `SELECT *`. But, there are definite readability and maintenance gains to being explicit. And, in the case of loading many columns into memory, there may be an application performance gain to being specific. I would re-word this a bit since saying "...when we can use an index" makes it seem like its a DB perf gain related to leveraging DB indexes.

Reply:
The sentence may not read well, but the idea is by replacing `SELECT *` for a `SELECT id, ...` we can leverage optimal indexes and have a performance gain, summing with the advantages you also mentioned. I'll try to work more on this, thank you.